packages/wrangler/src/api/dev.ts

```diff
@@ -307,21 +307,16 @@ export function parseRequestInput(
 ): [RequestInfo, RequestInit | undefined] {
 	if (input instanceof Request) {
 		return [input, undefined];
-	} else if (input instanceof URL) {
-		input = `${protocol}://${readyAddress}:${readyPort}${input.pathname}`;
-	} else if (typeof input === "string") {
-		try {
-			// Want to strip the URL to only get the pathname, but the user could pass in only the pathname
-			// Will error if we try and pass "/something" into new URL("/something")
-			input = `${protocol}://${readyAddress}:${readyPort}${
-				new URL(input).pathname
-			}`;
-		} catch {
-			input = `${protocol}://${readyAddress}:${readyPort}${input}`;
-		}
-	} else {
-		input = `${protocol}://${readyAddress}:${readyPort}`;
 	}
-
-	return [input, init];
+	if (!input) {
+		input = "/";
+	}
+	const url =
+		typeof input === "string"
+			? new URL(input, `${protocol}://${readyAddress}:${readyPort}`)
+			: input;
+	url.protocol = protocol;
+	url.hostname = readyAddress;
+	url.port = readyPort.toString();
```

Comment:
Let's avoid mutating the `input` object, just in case it is being reused elsewhere...
We can use the fact that `URL(url)` can convert `url` to a string.

```suggestion
	const url = new URL(input, `${protocol}://${readyAddress}:${readyPort}`);
	url.protocol = protocol;
	url.hostname = readyAddress;
	url.port = readyPort.toString();
```

Reply:
Ah, think I was working around this TS error:
> [ .                   ] src/api/dev.ts:311:22 - error TS2345: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
[ .                   ]   Type 'URL' is not assignable to type 'string'.
[ .                   ] 
[ .                   ] 311  const url = new URL(input, `${protocol}://${readyAddress}:${readyPort}`);

Can appease TS by updating this to:
```js
	const url = new URL(`${input}`, `${protocol}://${readyAddress}:${readyPort}`);
```

Reply:
Cool. I forgot about that type constraint.